Source/AliveLibAO/Slig.cpp

```diff
@@ -2069,9 +2084,253 @@ void Slig::MoveOnLine_467490()
+    }
+}
+
 signed __int16 Slig::HandlePlayerControlled_4667B0()
 {
-    NOT_IMPLEMENTED();
+    FP kScaleGrid = ScaleToGridSize_41FA30(field_BC_sprite_scale);
+
+    auto pressed = sInputObject_5009E8.field_0_pads[sCurrentControllerIndex_5076B8].field_0_pressed;
+    if (pressed & sInputKey_Right_4C6590)
+    {
+        if (!field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+        {
+            if (pressed & sInputKey_Run_4C65A8)
+            {
+                field_B4_velx = kScaleGrid / FP_FromInteger(4);
+                field_FC_current_motion = eSligStates::State_3_StandToRun_469C00;
+                return 1;
+            }
+
+            if (!WallHit_401930(field_BC_sprite_scale * FP_FromInteger(35), kScaleGrid * FP_FromInteger(2)))
+            {
+                field_B4_velx = kScaleGrid / FP_FromInteger(9);
+                field_FC_current_motion = eSligStates::State_16_StandingToStep_468FD0;
+                return 1;
+            }
+
+            if (WallHit_401930(field_BC_sprite_scale * FP_FromInteger(35), kScaleGrid))
+            {
+                return 0;
+            }
+            else
+            {
+                field_B4_velx = kScaleGrid / FP_FromInteger(6);
+                field_FC_current_motion = eSligStates::State_15_SteppingToStand_469080;
+                return 1;
+            }
+        }
+        else
+        {
+            field_FC_current_motion = eSligStates::State_5_TurnAroundStanding_469C80;
+            return 1;
+        }
+    }
+
+    if (pressed & sInputKey_Left_4C6594)
+    {
+        if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+        {
+            if (pressed & sInputKey_Run_4C65A8)
+            {
+                field_FC_current_motion = eSligStates::State_3_StandToRun_469C00;
+                field_B4_velx = -(kScaleGrid / FP_FromInteger(4));
+                return 1;
+            }
+            if (!WallHit_401930(field_BC_sprite_scale * FP_FromInteger(35), -kScaleGrid * FP_FromInteger(2)))
+            {
+                field_FC_current_motion = eSligStates::State_16_StandingToStep_468FD0;
+                field_B4_velx = -(kScaleGrid / FP_FromInteger(9));
+                return 1;
+            }
+            if (WallHit_401930(field_BC_sprite_scale * FP_FromInteger(35), -kScaleGrid))
+            {
+                return 0;
+            }
+            else
+            {
+                field_FC_current_motion = eSligStates::State_15_SteppingToStand_469080;
+                field_B4_velx = -(kScaleGrid / FP_FromInteger(6));
+                return 1;
+            }
+        }
+        else
+        {
+            field_FC_current_motion = eSligStates::State_5_TurnAroundStanding_469C80;
+            return 1;
+        }
+    }
+
+    if (sInputObject_5009E8.isHeld(sInputKey_DoAction_4C65A4)
+        && !(pressed & (sInputKey_LeftGameSpeakEnabler_4C65B8 | sInputKey_RightGameSpeakEnabler_4C65DC)))
+    {
+        if (Input_IsChanting_4334C0() == FALSE)
```

Comment:
I think everything else is using !Input_IsChanting_4334C0()

Reply:
Shouldn't we use this as it's more explicit, and this function doesn't return proper `bool`? (yet)

Reply:
shouldn't matter as !bool !int !short all convert to a boolean expression so it would need less changes for int -> bool this way